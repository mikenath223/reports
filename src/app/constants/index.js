export const pieColors = ["#A259FF", "#F24E1E", "#FFC107", "#6497B1"];
