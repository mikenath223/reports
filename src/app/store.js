import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import projectsReducer from "./slices/projectsSlice";
import formReducer from "./slices/formSlice";
import userReducer from "./slices/userSlice";
import gatewayReducer from "./slices/gatewaySlice";
import reportsReducer from "./slices/reportsSlice";

const middlewares = [];
if (process.env.NODE_ENV === `development`) {
  middlewares.push(logger);
}

export const store = configureStore({
  reducer: {
    projects: projectsReducer,
    form: formReducer,
    users: userReducer,
    gateways: gatewayReducer,
    reports: reportsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(middlewares),
});
