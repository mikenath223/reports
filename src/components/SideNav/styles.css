.side-nav {
  padding-left: 2rem;
}
