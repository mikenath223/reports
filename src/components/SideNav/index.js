import { ReactComponent as SideImage } from "app/images/sidenav.svg";
import "./styles.css";

export default function SideNav() {
  return (
    <nav className="side-nav">
      <SideImage />
    </nav>
  );
}
