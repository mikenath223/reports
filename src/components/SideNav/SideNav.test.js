import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "app/store";
import SideNav from ".";

describe("Sidenav", function () {
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <SideNav />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
