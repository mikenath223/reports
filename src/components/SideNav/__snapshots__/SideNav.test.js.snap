// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidenav should match snapshot 1`] = `
<div>
  <nav
    class="side-nav"
  >
    <svg>
      sidenav.svg
    </svg>
  </nav>
</div>
`;
